### DISCUSSION TITLE
Plugin Issue - netbox-topology-views and v4.0.2 / 4.0.8

### DISCUSSION THREAD

**Discussion by dan2522** [netbox-community/netbox#17110] [Help Wanted!]
I am trying to add the network-topology-views plugin, however during the migrate phase I am seeing the below issue:

SystemCheckError: System check identified some issues:

ERRORS:
vpn.IKEPolicy.mode: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.authentication_method: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IKEProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProfile.mode: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProposal.authentication_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.
vpn.IPSecProposal.encryption_algorithm: (fields.E120) CharFields must define a 'max_length' attribute.

I have tried this on version 4.0.2 and 4.0.8, and have the same issue on both versions

**Comment by bogdancordos** [comment]
Hi, this doesn't look like a plugin issue, more likely a custom filed on your VPN model.

**Reply by dan2522** [reply]
We do not have any custom fields in the VPN model, and are not using it at all

**Comment by candlerb** [comment]
What version of postgres are you running?

**Reply by dan2522** [reply]
psql (PostgreSQL) 14.12 (Ubuntu 14.12-0ubuntu0.22.04.1)

**Reply by candlerb** [reply]
Are you definitely running Netbox inside its own virtualenv, not using system-installed packages? Are you using the standard `./upgrade.sh` rather than running migrations manually?

**Reply by dan2522** [reply]
yes running in venv and upgraded from 3.5.8 -> 3.7.8 ->4.0.3 using ./upgrade.sh

**Reply by candlerb** [reply]
> vpn.IKEPolicy.mode: (fields.E120) CharFields must define a 'max_length' attribute.

The Netbox migrations are full of CharFields without a max_length attribute, so I think something is broken in terms of the software you've installed.

Disable the netbox_topology_views plugin (and all other plugins you have), remove it from local_requirements.txt, and run upgrade.sh again.  If everything migrates successfully, then the problem is with that plugin.  Contact author for assistance.

If you have this problem with *pure* Netbox, then you might have something in requirements.txt or local_requirements.txt which is pulling in a different version of Django than the one Netbox requires; but this is a problem you'll have to investigate yourself, as no-one else has experienced it AFAIK.

**Reply by dreng** [reply]
@dan2522 Have you made any progress on this? Based on the error message, it is very unlikely that the problem has to do with NetBox Topology Views. We do not use VPN in the plugin and have nothing to do with its migrations.

**Reply by dan2522** [reply]
Afternoon, yes managed to get to the bottom of this and the plugin installed. It ended up being to do with permissions and some of the previous netbox install being done under the root user and some a standard user.
